### DISCUSSION TITLE
"Outdoor mode" or "Family mode"

### DISCUSSION THREAD

**Discussion by developomp** [ppy/osu#13333] [Ideas] [answered]
- hides background images anywhere (hiding anime lolis)
- overrides the seasonal background setting

**Comment by NafeeJ** [comment]
Maybe just disable backgrounds? I feel like osu doesn't officially see itself as an anime game even though that just happens to be the case

**Comment by sr229** [comment]
There's really no point for this even if you do. It'll just break the UX for most beatmaps, and it's not really ideal as well.

**Comment by peppy** [comment, marked as answer]
An explicit content toggle is coming in the near future to the game itself, if that helps your scenario.

**Reply by vatei** [reply]
Bumping for visibility, It's a good QoL change and it might be important now that a bit more streamers are using lazer on stream

**Reply by Siiiuuuuu** [reply]
3 years ago and still not in 😭 😭 doesnt this take an 1 hour max to code in (i have no idea about coding but id imagine it wouldnt take long)

**Reply by peppy** [reply]
> 3 years ago and still not in 😭 😭 doesnt this take an 1 hour max to code in (i have no idea about coding but id imagine it wouldnt take long)

Explicit flag is not yet conveyed to the client. It would need to be fetched as part of the import process, potentially updated if it changes server-side, cached locally and then used for display purposes in a number of different places. It's a multi-day change involving multiple teams making changes.

**Reply by Siiiuuuuu** [reply]
> > 3 years ago and still not in 😭 😭 doesnt this take an 1 hour max to code in (i have no idea about coding but id imagine it wouldnt take long)
> 
> Explicit flag is not yet conveyed to the client. It would need to be fetched as part of the import process, potentially updated if it changes server-side, cached locally and then used for display purposes in a number of different places. It's a multi-day change involving multiple teams making changes.

Uhm... i was just thinking of the option to turn off all background and preview images in song select and when playing.. i didnt really mean the larger "family mode". (sorry for misreading and being too hasty on my side)

**Reply by developomp** [reply]
I agree with disabling all background idea. Displaying anime girl is okay in places like Japan and most western countries, but not so much in other parts of Asia or Middle east.

**Reply by monochrome22** [reply]
I assume the developers don't want to add an option to hide all backgrounds, since that would take value away from maps.
So as a middle ground, you could add the option to hide all backgrounds, but make it not save. When enabled, display the yellow text "This setting will revert upon exiting the game".
This way, the option to hide backgrounds will be there if someone needs it, but players will be discouraged from using it.
The explicit tag would likely not solve the issues some people have with backgrounds, since what counts as explicit is subjective.

**Reply by developomp** [reply]
@monochrome22 that's terrible UX

**Reply by monochrome22** [reply]
I agree it's not ideal, though I wouldn't call it terrible. 
I would also like to have a setting to hide all backgrounds, but I highly doubt peppy would agree to just add it. A middle ground solution is the only way it has at least a low chance of being added.
